src/internal.ts

```diff
@@ -163,7 +171,9 @@ export const constructBalanceModelTx = async (
   txFields?: BalanceModelTransactionFields
 ): Promise<ConstructedTransactionResponse> => {
   // TODO: cache block height somewhere.
-  const { BlockHeight } = await getAppState();
+  const { BlockHeight } = await (txFields?.Nonce
+    ? Promise.resolve({ BlockHeight: undefined })
```

Comment:
per above comment I think we could remove the awkward `BlockHeight: undefined` here if we just rely on the existence of the `Nonce` option.